charts/external-dns/templates/deployment.yaml

```diff
@@ -132,6 +136,47 @@ spec:
           resources:
             {{- toYaml . | nindent 12 }}
           {{- end }}
+        {{- if eq $provider "webhook" }}
+        {{- with .Values.provider }}
+        - name: {{ $providername }}
+          image: {{ get $whp .name }}
+          imagePullPolicy: {{ $.Values.image.pullPolicy }}
+          {{- with .env }}
+          env:
+            {{- toYaml . | nindent 12 }}
+          {{- end }}
+          {{- with .args }}
+          args:
+            - {{ toYaml . | nindent 12 }}
+          {{- end }}
+          ports:
+            - name: http
+              protocol: TCP
+              containerPort: 8888
+          livenessProbe:
+            {{- toYaml $.Values.livenessProbe | nindent 12 }}
+          readinessProbe:
+            {{- toYaml $.Values.readinessProbe | nindent 12 }}
```

Comment:
Sharing livenessProbe and readinessProbe seems inadvisable, especially since the ports are going to be different.

Reply:
The probes are shared for external-dns and also in current known webhook provider implementations.
As a matter of fact, using named port, it's working like a charm :). 
The `http` port 8888 is used on webhook sidecar for probes and the `http` port 7979 is used for probes on external-dns.